miden-lib/asm/sat/layout.masm

```diff
@@ -705,6 +735,17 @@ export.get_created_note_num_assets
     push.CREATED_NOTE_METADATA_OFFSET add mem_load
 end
 
+#! Sets the number of assets in the created note
+#!
+#! Stack: [note_ptr, num_assets]
+#! Output: []
+#!
+#! - note_ptr is the memory address at which the created note data begins.
+#! - num_assets is the number of assets in the created note.
+export.set_created_note_num_assets
+    push.CREATED_NOTE_METADATA_OFFSET add mem_store
```

Comment:
Should the check of whether `num_assets` is within max note asset limit happen here?

Reply:
I have tried to ensure the `layout` module has a single responsibility of loading and storing data to / from memeory.  As such the `layout` module does not contain any logic to interpret the data and it is the responsility of the caller to ensure the value being stored is valid. In my mind this creates a clean seperation of concerns.

Reply:
Agreed! Let's keep it the way you have it.